File: app/src/main/java/knf/kuma/database/dao/SeeingDAO.java
package knf.kuma.database.dao;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.TypeConverters;
import androidx.room.Update;
import knf.kuma.database.BaseConverter;
import knf.kuma.pojos.SeeingObject;

@Dao
@TypeConverters(BaseConverter.class)
public interface SeeingDAO {

    @Query("SELECT * FROM seeingobject ORDER BY title")
    LiveData<List<SeeingObject>> getAll();

    @Query("SELECT * FROM seeingobject ORDER BY title")
    List<SeeingObject> getAllRaw();

    @Query("SELECT * FROM seeingobject WHERE aid LIKE :aid")
    SeeingObject getByAid(String aid);

    @Query("SELECT count(*) FROM seeingobject WHERE aid LIKE :aid")
    Boolean isSeeing(String aid);

    @Query("SELECT count(*) FROM seeingobject")
    LiveData<Integer> getCountLive();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void add(SeeingObject object);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void addAll(List<SeeingObject> object);

    @Update
    void update(SeeingObject object);

    @Delete
    void remove(SeeingObject object);

    @Query("DELETE FROM seeingobject")
    void clear();
}

File: app/src/main/java/knf/kuma/database/dao/SeeingDAO.kt
package knf.kuma.database.dao

import androidx.lifecycle.LiveData
import androidx.paging.DataSource
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.TypeConverters
import androidx.room.Update
import knf.kuma.database.BaseConverter
import knf.kuma.pojos.SeeingObject

@Dao
@TypeConverters(BaseConverter::class)
interface SeeingDAO {
    @get:Query("SELECT * FROM seeingobject ORDER BY title")
    val allRaw: MutableList<SeeingObject>
    @Query("SELECT * FROM seeingobject WHERE aid=:aid")
    fun findByAid(aid: String): SeeingObject?
    @get:Query("SELECT count(*) FROM seeingobject")
    val countAll: Int
    @Delete
    fun remove(seeingObject: SeeingObject)
}
